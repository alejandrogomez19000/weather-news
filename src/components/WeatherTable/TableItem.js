import React from "react";
import "./WeatherTable.scss";
import propTypes from "prop-types";

const TableItem = ({
  maxTemp,
  minTemp,
  day,
  date,
  description,
  humidity,
  icon,
}) => (
  <div className="table-item">
    <div className="date-container">
      <span>{day}</span>
      <span className="date">{date}</span>
    </div>
    <img src={icon} className="table-item-icon" alt="weather" />
    <div className="temp-container">
      <span className="max-temp">{maxTemp}Â°</span>
      <span>/{minTemp}</span>
    </div>
    <div className="description">{description}</div>
    <div className="humidity">
      <span>Hum.</span>
      {humidity}%
    </div>
  </div>
);

TableItem.propTypes = {
  maxTemp: propTypes.number.isRequired,
  minTemp: propTypes.number.isRequired,
  day: propTypes.string.isRequired,
  date: propTypes.string.isRequired,
  description: propTypes.string.isRequired,
  humidity: propTypes.number.isRequired,
  icon: propTypes.string.isRequired,
};

export default TableItem;
