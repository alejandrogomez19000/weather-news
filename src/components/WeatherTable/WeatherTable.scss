.tab-content {
  .weather-content {
    padding: 20;
    display: flex;
    flex-direction: column;
    .content-list {
      padding: 0;
      margin: 0;

      .table-item {
        display: flex;
        padding: 15px;
        border: 1px solid #e5e5e5;
        align-items: center;
        height: 5rem;
        .temp-container {
          font-size: 0.9rem;
          padding: 0.5rem;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          margin-right: 1rem;
          width: 30%;
          span {
            color: #999;
            font-weight: 300;
          }
          .max-temp {
            font-size: 1.1rem;
            color: #222;
            font-weight: 500;
          }
        }

        .table-item-icon {
          height: 4rem;
          width: 4rem;
        }
        .description {
          flex: 1;
          height: 100%;
          display: flex;
          align-items: center;
          text-transform: capitalize;
        }
        .date-container {
          display: flex;
          flex-direction: column;
          align-items: center;
          .date {
            color: #777171;
          }
        }
        .humidity {
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
          color: #777171;
          padding: 0.5rem;
        }
      }
    }
  }
}

@media (max-width: 600px) {
  body {
    font-size: 14px !important;
  }
}

@media (max-width: 400px) {
  body {
    font-size: 11px !important;
  }
}
