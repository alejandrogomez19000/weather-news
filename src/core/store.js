import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import sagas from "core/sagasRoot";
import reducers from "core/reducerRoot";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(reducers, compose(applyMiddleware(sagaMiddleware)));
sagaMiddleware.run(sagas);

export default store;
