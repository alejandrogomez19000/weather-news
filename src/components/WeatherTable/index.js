import React from "react";
import propTypes from "prop-types";
import "./WeatherTable.scss";
import TableItem from "./TableItem";
import { parseUnixDate, getDay, weekDays } from "utils/helper";
import { v4 as uuidv4 } from "uuid";

const WeatherTable = ({ data }) => {
  const renderTable = () =>
    data?.map((v, k) => {
      if (k < 5) {
        const { max, min } = data[k].temp;
        const { icon, description } = data[k].weather[0];
        return (
          <TableItem
            key={uuidv4()}
            day={weekDays[getDay(v.dt)]}
            date={parseUnixDate(v.dt)}
            maxTemp={Math.round(max * 10) / 10}
            minTemp={Math.round(min * 10) / 10}
            description={description}
            humidity={data[k].humidity}
            icon={`http://openweathermap.org/img/wn/${icon}@2x.png`}
          />
        );
      }
    });

  return (
    <div className="tab-content">
      <div className="weather-content">
        <ul className="content-list">{renderTable()}</ul>
      </div>
    </div>
  );
};

WeatherTable.propTypes = {
  data: propTypes.arrayOf(propTypes.any).isRequired,
};

export default WeatherTable;
