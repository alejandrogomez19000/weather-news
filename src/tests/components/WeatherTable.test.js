import React from "react";
import WeatherTable from "components/WeatherTable";
import ShallowRender from "react-test-renderer/shallow";
import { Provider } from "react-redux";
import store from "core/store";

const renderer = new ShallowRender();

const data = [
  {
    weather: [
      {
        description: "some desc",
        icon: "some icon",
      },
    ],
    temp: {
      max: 10,
      min: 5,
    },
    dt: "159258924248",
    humidity: 23,
  },
];

describe("render component", () => {
  it("should be render", async () => {
    renderer.render(
      <Provider store={store}>
        <WeatherTable data={data} />
      </Provider>
    );
    const renderOutput = renderer.getRenderOutput();
    expect(renderOutput).toMatchSnapshot();
  });
});
