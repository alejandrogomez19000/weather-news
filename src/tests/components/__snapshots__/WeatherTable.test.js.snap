// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component should be render 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <WeatherTable
    data={
      Array [
        Object {
          "dt": "159258924248",
          "humidity": 23,
          "temp": Object {
            "max": 10,
            "min": 5,
          },
          "weather": Array [
            Object {
              "description": "some desc",
              "icon": "some icon",
            },
          ],
        },
      ]
    }
  />
</Context.Provider>
`;
